packages/transactions/src/authorization.ts

```diff
@@ -503,177 +493,141 @@ function verifySingleSig(
   return nextSigHash;
 }
 
-export class Authorization extends Deserializable {
-  authType?: AuthType;
-  spendingCondition?: SpendingCondition;
-  sponsorSpendingCondition?: SpendingCondition;
-
-  constructor(
-    authType?: AuthType,
-    spendingConditions?: SpendingConditionOpts,
-    sponsorSpendingCondition?: SpendingConditionOpts
-  ) {
-    super();
-    this.authType = authType;
-    if (spendingConditions) {
-      this.spendingCondition = {
-        ...spendingConditions,
-        nonce: intToBigInt(spendingConditions.nonce, false),
-        fee: intToBigInt(spendingConditions.fee, false),
-      };
-    }
-    if (sponsorSpendingCondition) {
-      this.sponsorSpendingCondition = {
-        ...sponsorSpendingCondition,
-        nonce: intToBigInt(sponsorSpendingCondition.nonce, false),
-        fee: intToBigInt(sponsorSpendingCondition.fee, false),
-      };
-    }
-  }
+export type Authorization = StandardAuthorization | SponsoredAuthorization;
 
-  intoInitialSighashAuth(): Authorization {
-    if (this.spendingCondition) {
-      switch (this.authType) {
-        case AuthType.Standard:
-          return new Authorization(AuthType.Standard, clearCondition(this.spendingCondition));
-        case AuthType.Sponsored:
-          return new Authorization(
-            AuthType.Sponsored,
-            clearCondition(this.spendingCondition),
-            newInitialSigHash()
-          );
-        default:
-          throw new SigningError('Unexpected authorization type for signing');
-      }
-    }
+export interface StandardAuthorization {
+  authType: AuthType.Standard;
+  spendingCondition: SpendingCondition;
+}
 
-    throw new Error('Authorization missing SpendingCondition');
-  }
+export interface SponsoredAuthorization {
+  authType: AuthType.Sponsored;
+  spendingCondition: SpendingCondition;
+  sponsorSpendingCondition: SpendingCondition;
+}
 
-  setFee(amount: IntegerType) {
-    switch (this.authType) {
-      case AuthType.Standard:
-        this.spendingCondition!.fee = intToBigInt(amount, false);
-        break;
-      case AuthType.Sponsored:
-        this.sponsorSpendingCondition!.fee = intToBigInt(amount, false);
-        break;
-    }
+export function createStandardAuth(spendingCondition: SpendingCondition): StandardAuthorization {
+  return {
+    authType: AuthType.Standard,
+    spendingCondition,
+  };
+}
+
+export function createSponsoredAuth(
+  spendingCondition: SpendingCondition,
+  sponsorSpendingCondition?: SpendingCondition
+): Authorization {
+  if (!sponsorSpendingCondition) {
+    sponsorSpendingCondition = createSingleSigSpendingCondition(
```

Comment:
I'd write this in a way where you don't reassign the parameter. Perhaps do this conditional as a ternary in the return object itself?